app/containers/TracksContainer/saga.js

```diff
@@ -10,9 +10,11 @@ export function* requestGetTracks(action) {
   const res = yield call(getSongs, action.trackName);
   const { data, ok } = res;
 
+  const tracksDataResponse = data?.results;
+  const optimisedTrackData = tracksDataResponse?.reduce((obj, item) => ({ ...obj, [item.trackId]: { ...item } }), []);
```

Comment:
Why is the initial value of obj set to [] ?

Reply:
it's a typo, should be empty object.